sys/kern/vmem.c

```diff
@@ -0,0 +1,140 @@
+#include <sys/vmem.h>
+#include <sys/queue.h>
+#include <sys/pool.h>
+#include <sys/mimiker.h>
+#include <sys/libkern.h>
+
+#define VMEM_MAXORDER ((int)(sizeof(vmem_size_t) * CHAR_BIT))
+#define VMEM_MAXHASH 512
+#define VMEM_NAME_MAX 16
+
+#define VMEM_ADDR_MIN 0
+#define VMEM_ADDR_MAX (~(vmem_addr_t)0)
+
+#define SIZE2ORDER(size) ((int)log2(size))
+
+typedef TAILQ_HEAD(vmem_seglist, bt) vmem_seglist_t;
+typedef LIST_HEAD(vmem_freelist, bt) vmem_freelist_t;
+typedef LIST_HEAD(vmem_hashlist, bt) vmem_hashlist_t;
+
+static LIST_HEAD(, vmem) vmem_list = LIST_HEAD_INITIALIZER(vmem_list);
+
+typedef struct vmem {
+  vm_flag_t vm_flags;
+  size_t vm_size;
+  size_t vm_inuse;
+  char vm_name[VMEM_NAME_MAX];
+  vmem_seglist_t vm_seglist;
+  vmem_freelist_t vm_freelist[VMEM_MAXORDER];
+  vmem_hashlist_t vm_hashlist[VMEM_MAXHASH];
+  LIST_ENTRY(vmem) vm_alllist;
+} vmem_t;
+
+typedef enum { BT_TYPE_FREE, BT_TYPE_BUSY, BT_TYPE_SPAN } bt_type_t;
+
+typedef struct bt {
+  TAILQ_ENTRY(bt) bt_seglist;
```

Comment:
If that's an entry on list, then it's called `link`. Unfortunately this convention is not used throughout whole code base.

Please rename to `bt_seglink` and perform similar name transformation for other links.

Reply:
Thanks, done!